### DISCUSSION TITLE
Valid items in editable dialog box configuration

### DISCUSSION THREAD

**Discussion by zangab** [pimcore/pimcore#9931] [Q&A / Support]
Hi,
is there a list of items, which are valid in the `EditableDialogBoxConfiguration`?
I thought I could use all kinds of `types` that appear in the data-object class definition (layouts & data components). Turns out, this is not the case. Or maybe I did something wrong. However, for fieldset, the "No loader for fieldset" error appears. Where can I find any documentation on this or list of valid items??

In the pimcore-demo, it uses `tabpanels` and `panels`, see [Brick ImageMetadata](https://github.com/pimcore/demo/blob/10.x/src/Document/Areabrick/ImageMetadata.php).

Thx.

**UPDATE**
The fieldset started to work somehow (can't tell why) but the `text` layout component that displays html text is also an issue.

**ANOTHER UPDATE**
The config of the layout elements (like the `border` of a panel) is also a blackbox. No info whatsoever..... Rly disappointing imo.

```php
// this is my item in a 'panel'
[
    'type' => 'text',
    'name' => 'checkbox_infotext',
    'title' => '',
    'html' => "This is my <strong>superb</strong> HTML output",
],
```

**Comment by NiklasBr** [comment]
I'm also looking for additional documentation on this, it's a lot of trial-and-error and its never really clear if the behaviour is intended or not. As an example, how to add a default value to a `numeric` field? All I know is that neither of these works and I can't find any documentation:

```php
                        [
                            'type' => 'numeric',
                            'label' => $this->translate('brick.options.show-count'),
                            'name' => 'show_count',
                            'config' => [
                                'value' => 1,
                                'default' => 2,
                                'defaultValue' => 3,
                            ],
                            'value' => 4,
                            'default' => 5,
                            'defaultValue' => 6,
                        ],
```

Is it a bug? How should I know when documentation is missing!

**Reply by alexej-d** [reply]
You can only use editables for documents, not for date objects. The possible editable types can be found [here](https://pimcore.com/docs/pimcore/current/Development_Documentation/Documents/Editables/) (click through the types in the left list). This ist also where you can find the possible configuration parameters.

**Reply by zangab** [reply]
@alexej-d 
> You can only use editables for documents...

that is just wrong. if u check the example with the brick imagemetadata, you will see that u can also set layouts.

@NiklasBr 
editable config should be in the `config` key. not sure about the default-value though as this is a "specific" setting according to the data-object class builder UI. but in there, they set it with the key `defaultValue` 🤔

**Reply by alexej-d** [reply]
@zangab I was only speaking about possible data fields, not layout elements. I tried to clear a possible misunderstanding the thread starter might have as he mentioned data objects.

My statement still stands. Apart from document editable data types, you can also use some layout elements which happen to have an intersection with layout types of data objects.

**Reply by NiklasBr** [reply]
@alexej-d Do you know of any documentation regarding it?

**Reply by alexej-d** [reply]
@NiklasBr only the documentation you can find [here](https://pimcore.com/docs/pimcore/current/Development_Documentation/Documents/Editables/Areablock/Bricks.html#page_Editable-Dialog-since-6-8)

**Reply by NiklasBr** [reply]
Yeah, it's definitely lacking :(

**Comment by M-Ali-Aziz** [comment]
Hi,
I have issue with `Editable Dialog` and I hope you can help me.
The params data exist only on the editmode(when loged in on Admin page) and it is `Null` on frontend.

I set param on my `action` function on the Areabrick:
```
$value = $this->getName(); // string
$info->setParam('name', $value);
```

on view.html.php:
`$this->name = null `
???!!!

I don't know why the params data gets null on the frontend !?

Any help please!

**Reply by dvesh3** [reply]
@M-Ali-Aziz I see you are using `.html.php` template, so are you using Pimcore 6 or 10 with php templating bundle? if later is the case then please have a look at the recent fix in the core 10.1.3 https://github.com/pimcore/pimcore/commit/7e697ae2288ef8de9f9ab6d2331d4872e52f38e7

**Reply by M-Ali-Aziz** [reply]
Thank you for your reply.
I'm using Pimcore 6.

**Reply by dvesh3** [reply]
@M-Ali-Aziz sorry, Pimcore 6 reached it EOL, so no further fixes in community edition.

**Reply by M-Ali-Aziz** [reply]
OK, thanks any way. 
I'm using the old Configuration in Editmide (edit.html.php) method and it's working fine.

I'm doing my best to update this project to Pimcore 10 as soon as possible.

**Comment by patric-eberle** [comment]
This is also unclear for us. It would be helpful to use such layout options (as on classes), but it is not clear what and how they can be used. I can confirm that `tabpanel`, `panel`, `fieldset` and `fieldcontainer` seem to work. But are they save to use? And are there additional options that just `title` and `items`?
